brax/envs/to_torch.py

```diff
@@ -0,0 +1,133 @@
+""" Wrapper around a Brax GymWrapper, that converts its outputs to PyTorch tensors.
+
+This conversion happens direcly on-device, witout the need to move values to the CPU.
+"""
+
+import datetime
+import time
+from collections import abc
+from functools import partial, singledispatch
+from typing import Any, Dict, Mapping, Optional, TypeVar, Union
+
+import gym
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from jax._src import dlpack as jax_dlpack
+from jaxlib.xla_extension import DeviceArray
+from torch import Tensor
+from torch.utils import dlpack as torch_dlpack
+
+from brax.envs.wrappers import GymWrapper, VectorGymWrapper
+
+
+class JaxToTorchWrapper(gym.Wrapper):
+    """Wrapper that converts Jax tensors to PyTorch tensors."""
+
+    def __init__(self, env: Union[GymWrapper, VectorGymWrapper], device: torch.device = None):
+        """Creates a Wrapper around a `GymWrapper` or `VectorGymWrapper` so it outputs
+        PyTorch tensors.
+
+        Parameters
+        ----------
+        env : Union[GymWrapper, VectorGymWrapper]
+            A `GymWrapper` or `VectorGymWrapper` to wrap.
+        device : torch.device, optional
+            device on which to move the Tensors. Defaults to `None`, in which case the
+            tensors will be on the same devices as their Jax equivalents.
+        """
+        if not hasattr(env, "reward_range"):
+            # NOTE: Fix a bug with the current GymWrapper, where the `Env` doesn't have
+            # the `reward_range` attribute which is needed by the gym.Wrapper
+            # constructor.
+            env.reward_range = (-np.inf, np.inf)
+        super().__init__(env)
+        self.device: Optional[torch.device] = device
+
+    def observation(self, observation):
+        return jax_to_torch(observation, device=self.device)
+
+    def action(self, action):
+        return torch_to_jax(action)
+
+    def reward(self, reward):
+        return jax_to_torch(reward, device=self.device)
+
+    def done(self, done: DeviceArray) -> Tensor:
+        return jax_to_torch(done, device=self.device)
+
+    def info(self, info: Dict) -> Dict:
+        return jax_to_torch(info, device=self.device)
+
+    def reset(self):
+        obs = super().reset()
+        return self.observation(obs)
+
+    def step(self, action):
+        action = self.action(action)
+        obs, rewards, done, info = super().step(action)
+        obs = self.observation(obs)
+        rewards = self.reward(rewards)
+        done = self.done(done)
+        info = self.info(info)
+        return obs, rewards, done, info
+
+
+@singledispatch
```

Comment:
These functions are cool and may be generally useful elsewhere!  Can you move them into something like `brax/io/torch.py`, and then put the wrapper into `wrappers.py`?

Reply:
> an you move them into something like brax/io/torch.py

Absolutely!

>  and then put the wrapper into wrappers.py?

I don't think this is a good idea: it would make `torch` a mandatory requirement for using anything in `wrappers.py`!

Reply:
Perhaps there could be a `wrappers` folder that would contain the contents of `wrappers.py`, renamed to something like `base.py`, and then this `to_torch.py` file would be placed in there? What do you think?
This would be done in another PR if you want.

Reply:
Oops - I missed this.  Yes, good point.  We can save that for some other time - leaving it as is for now works for me.